### PR TITLE
39 make robot drive to goal to deliver balls

### PR DESCRIPTION
**Pull request by yech199**
Routine for delivering ball to goal. Some offsets might be off and should be corrected, but this is a general issue :sweat_smile:

### CODE REVIEW

File: remote/src/main/java/courseObjects/Border.java [hunk 1]
```diff
@@ -54,4 +57,61 @@ public synchronized void setTopLeft(Point topLeft) {
...
+    /**
+     * Update the small goal point before using it
+     */
+    private synchronized boolean updateSmallGoalPoint() {
+        // Can't calculate point without these
+        if (this.topLeft == null || this.topRight == null || this.bottomLeft == null || this.bottomRight == null)
+            return false;
+
+        // Set smallGoalPos only once (left or right)
+        if (this.smallGoalPos == null)
```
Comment (Greve2001) [inline remote/src/main/java/courseObjects/Border.java:70]: I think a better name for this, would be "direction"

File: remote/src/main/java/courseObjects/Border.java [hunk 2]
```diff
@@ -54,4 +57,61 @@ public synchronized void setTopLeft(Point topLeft) {
...
+    /**
+     * NB! Always call getSmallGoalPoint before this one to makes sure x and y of smallGoalMiddlePoint is set correctly
+     * @return The point to which 'the rear end of the robot' should arrive before turning toward the goal with an angle
+     */
+    public synchronized Point getSmalGoalDestPoint() {
+        if (!updateSmallGoalPoint())
+            return null;
+
+        // The distance between the goal and the point where the rear end of the robot should stop
+        int offset = 25;
```
Comment (Greve2001) [inline remote/src/main/java/courseObjects/Border.java:105]: Should this not change, depending on the robots length?

File: remote/src/main/java/vision/Algorithms.java [hunk 3]
```diff
@@ -51,22 +51,51 @@ public static Ball findClosestBall(List<Ball> balls, Robot robot) {
...
     }
 
+    /**
+     * Finds the shortest angle between robot and ball.
+     *
+     * @return angle in degrees. Clockwise with the robot returns positive angle-values, counter-clockwise with the robot returns negative angle-values.
+     */
+    public static double findRobotShortestAngleToBall(Robot robot, Ball ball) {
+        return findRobotShortestAngleToPoint(robot, ball.getCenter());
+    }
```
Comment (Greve2001) [inline remote/src/main/java/vision/Algorithms.java:81]: Should we not just remove this wrapper, and just call the inner method, the other places

**Review by Greve2001 — CHANGES_REQUESTED** [review]
I have commented the code you have made. I have not commented on the parts from Mikael's branch, as that needs to be merged first

**Review by MikaelFangel — APPROVED** [review]

**Review by MikaelFangel — APPROVED** [review]

**Pull request merged**